Fix -Wimplicit-function-declaration error with gcc 14.

Thanks to Ariadne Conill for the suggested fix.

Example error output:

```
In file included from ../src/asus.h:6,
                 from ../src/libratbag-data.c:33:
../src/libratbag-data.c: In function 'file_data_matches':
../src/libratbag-data.c:610:66: error: implicit declaration of function
'basename'; did you mean 'g_basename'? [-Wimplicit-function-declaration]
  610 |                 log_error(ratbag, "Missing DeviceMatch in %s\n", basename(path));
      |                                                                  ^~~~~~~~
```

--- libratbag-0.17-origin/src/libratbag-data.c
+++ libratbag-0.17/src/libratbag-data.c
@@ -607,7 +607,7 @@
 
 	match_strv = g_key_file_get_string_list(keyfile, GROUP_DEVICE, "DeviceMatch", NULL, NULL);
 	if (!match_strv) {
-		log_error(ratbag, "Missing DeviceMatch in %s\n", basename(path));
+		log_error(ratbag, "Missing DeviceMatch in %s\n", g_basename(path));
 		return false;
 	}
 
@@ -623,7 +623,7 @@
 
 	data->driver = g_key_file_get_string(keyfile, GROUP_DEVICE, "Driver", NULL);
 	if (!data->driver) {
-		log_error(ratbag, "Missing Driver in %s\n", basename(path));
+		log_error(ratbag, "Missing Driver in %s\n", g_basename(path));
 		return false;
 	} else {
 		const struct driver_map *map;
@@ -640,14 +640,14 @@
 
 		if (data->drivertype == NONE) {
 			log_error(ratbag, "Unknown driver %s in %s\n",
-				  data->driver, basename(path));
+				  data->driver, g_basename(path));
 			return false;
 		}
 	}
 
 	ledtypes_strv = g_key_file_get_string_list(keyfile, GROUP_DEVICE, "LedTypes", NULL, NULL);
 	if (parse_ledtypes(ledtypes_strv, data->led_types, ARRAY_LENGTH(data->led_types)) < 0) {
-		log_error(ratbag, "Invalid LedTypes string in '%s'\n", basename(path));
+		log_error(ratbag, "Invalid LedTypes string in '%s'\n", g_basename(path));
 		return false;
 	}
 
